src/main/java/org/myrobotlab/math/MapperLinear.java

```diff
@@ -0,0 +1,237 @@
+package org.myrobotlab.math;
+
+import java.io.Serializable;
+
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.logging.LoggingFactory;
+import org.myrobotlab.service.WorkETest;
+import org.slf4j.Logger;
+
+public final class MapperLinear implements Serializable, MapperInterface {
+
+  private static final long serialVersionUID = 1L;
+  public final static Logger log = LoggerFactory.getLogger(MapperLinear.class);
+
+  // mapping range related
+  Double minX;
+  Double maxX;
+  Double minY;
+  Double maxY;
+
+  // clipping
+  Double minOutput;
+  Double maxOutput;
+
+  Boolean inverted = false;
+  Double maxInput;
+  Double minInput;
+
+  /**
+   * non-parameterized contructor for "un-set" mapper
+   * use the merge function to set "default" values from the motorcontroller
+   */
+  public MapperLinear() {
+  }
+  
+  public MapperLinear(double minX, double maxX, double minY, double maxY) {
+    map(minX, maxX, minY, maxY); 
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#calcOutput(double)
+   */
+  @Override
+  final public double calcOutput(double in) {
+    if (minInput != null && in < minInput) {
+      log.warn("clipping input {} to {}", in, minInput);
+      in = minInput;
+    }
+    if (maxInput != null && in > maxInput) {
+      log.warn("clipping input {} to {}", in, maxInput);
+      in = maxInput;
+    }
+    
+    double c = minY + ((in - minX) * (maxY - minY)) / (maxX - minX);

```

Comment:
No protection against divide by zero here..

Reply:
Heh .. there is in the sense of a throw ;)
But sure I'll do a check with error event

Reply:
I think logging an error/warning  and ultimately, return "0.0"  ... in that it's *safe*.. if you divide by zero.. motor output should be 0.   you can't return null with this method signature..   (alternatively, we could throw an exception, but I feel like an uncaught exception like this breaks people scripts sometimes.)